### PR TITLE
Add support for reading VictorOps API key from file

### PR DESCRIPTION
**Pull request by OktarianTB**
Related to issue #2498. Basically a copy of https://github.com/prometheus/alertmanager/pull/2728.

https://github.com/prometheus/alertmanager/pull/2554 can be closed as a result.

Signed-off-by: Oktarian Tilney-Bassett 😄

### CODE REVIEW

File: docs/configuration.md [hunk 1]
```diff
@@ -975,6 +976,9 @@ VictorOps notifications are sent out via the [VictorOps API](https://help.victor
 # The API key to use when talking to the VictorOps API.
```
Comment (simonpasquier) [inline docs/configuration.md:976]:
```suggestion
# The API key to use when talking to the VictorOps API.
# It is mutually exclusive with `api_key_file`.
```

File: docs/configuration.md [hunk 2]
```diff
@@ -975,6 +976,9 @@ VictorOps notifications are sent out via the [VictorOps API](https://help.victor
...
 [ api_key: <secret> | default = global.victorops_api_key ]
 
+# The filepath to API key to use when talking to the VictorOps API. Conflicts with api_key.
```
Comment (simonpasquier) [inline docs/configuration.md:979]:
```suggestion
# Reads the API key to use when talking to the VictorOps API from a file.
# It is mutually exclusive with `api_key`.
```

File: notify/victorops/victorops.go [hunk 3]
```diff
@@ -70,7 +73,19 @@ func (n *Notifier) Notify(ctx context.Context, as ...*types.Alert) (bool, error)
 		tmpl   = notify.TmplText(n.tmpl, data, &err)
 		apiURL = n.conf.APIURL.Copy()
 	)
-	apiURL.Path += fmt.Sprintf("%s/%s", n.conf.APIKey, tmpl(n.conf.RoutingKey))
+
+	var apiKey string
+	if n.conf.APIKey != "" {
+		apiKey = string(n.conf.APIKey)
+	} else {
+		content, fileErr := os.ReadFile(n.conf.APIKeyFile)
```
Comment (simonpasquier) [inline notify/victorops/victorops.go:81]: I might miss the reason why we need a new variable name but if there's none, I'd prefer to keep `err` :) 
```suggestion
		content, err := os.ReadFile(n.conf.APIKeyFile)
```

**Review by simonpasquier — COMMENTED** [review]
Good! Just a few nits...

File: notify/victorops/victorops.go (hunk 3 above)
Comment (OktarianTB) [reply notify/victorops/victorops.go:81]: Declaring it as `err` would be a shadow declaration of a previously defined `err` variable used for templating errors: 🙂 
https://github.com/prometheus/alertmanager/blob/3f561cdce69ecff5cb7cff1a611b3fc592e964a9/notify/victorops/victorops.go#L70-L75

Comment (OktarianTB) [reply notify/victorops/victorops.go:81]: That makes sense! I agree that the templating error handling is not the best, just need to be careful I guess :)

Comment (OktarianTB) [reply notify/victorops/victorops.go:81]: Are there any other changes needed for this PR btw? 🏁

**Review by simonpasquier — APPROVED** [review]

**Pull request merged**